src/executor/column_options/auto_increment.rs

```diff
@@ -0,0 +1,96 @@
+#![cfg(feature = "auto-increment")]
+use {
+    crate::{
+        data::{Row, Value},
+        result::MutResult,
+        store::{AlterTable, AutoIncrement, Store, StoreMut},
+    },
+    futures::executor,
+    sqlparser::{
+        ast::{ColumnDef, ColumnOption, ColumnOptionDef},
+        dialect::keywords::Keyword,
+        tokenizer::{Token, Word},
+    },
+    std::fmt::Debug,
+};
+
+pub fn run<T: 'static + Debug, U: Store<T> + StoreMut<T> + AlterTable + AutoIncrement>(
+    storage: U,
+    rows: Vec<Row>,
+    column_defs: &[ColumnDef],
+    table_name: &str,
+) -> MutResult<U, Vec<Row>> {
+    let auto_increment_columns: Vec<(usize, &ColumnDef)> = column_defs
+        .iter()
+        .enumerate()
+        .filter(|(_, ColumnDef { options, .. })| {
+            options
+                .iter()
+                .find_map(|ColumnOptionDef { option, .. }| {
+                    if is_auto_increment(option) {
+                        Some(true)
+                    } else {
+                        None
+                    }
+                })
+                .unwrap_or(false)
+        })
+        .collect();
+
+    if !auto_increment_columns.is_empty() {
+        let mut new_rows: Vec<Row> = vec![];
+        let storage = rows.into_iter().fold(storage, |storage, row| {
+            let mut row = row.0;
+            let storage =
+                auto_increment_columns
+                    .iter()
+                    .fold(storage, |storage, (index, column)| {
+                        if matches!(row[*index], Value::Null) {
+                            let name = &column.name.value;
+
+                            match executor::block_on(storage.generate_value(table_name, name)) {
```

Comment:
You cannot use `futures::executor`, it is only supported in limited environments.
e.g. WebAssembly does not support this.

Reply:
What can I use instead?

Reply:
You don't need to block, you can simply use `async` :)

Reply:
Nevermind, it is only used in an async function. Not sure why I didn't do that before.